services/storehost/replicationJobRunner.go

```diff
@@ -224,6 +234,19 @@ func (runner *replicationJobRunner) run() {
 		}).Info(`replication for extent started`)
 	}
 
+	var maxConsecutiveFailures int
+	for existingFailedJob, failedTimes := range runner.failedJobs {
+		if _, ok := currentFailedJobs[existingFailedJob]; !ok {
+			delete(runner.failedJobs, existingFailedJob)
+		} else {
+			runner.failedJobs[existingFailedJob] = failedTimes + 1
+			if runner.failedJobs[existingFailedJob] > maxConsecutiveFailures {
+				maxConsecutiveFailures = runner.failedJobs[existingFailedJob]
+			}
+		}
+	}
+	runner.m3Client.UpdateGauge(metrics.ReplicateExtentScope, metrics.StorageReplicationJobMaxConsecutiveFailures, int64(maxConsecutiveFailures))
```

Comment:
also, would it makes sense to emit a simple counter on total number of job failures ? Or do we already have it ?